RPIPs/RPIP-draft.md

```diff
@@ -0,0 +1,95 @@
+  can submit prices at least once a week
+- In the medium term, the watchtower SHOULD implement a voting scheme that will show each oDAO
+  member can generate a correct merkle tree; this MAY have a moderate timeout to avoid degrading NO
+  experience much
+- The next significant smart contract release SHALL allow for voting past consensus
+
+## Reference Implementation
+### Subcommittees for balance and price duties
+This change can be implemented now in the watchtower.
+
+Each oDAO node figures out its own index (similar code already exists). On an even update
+`((current_update_block - first_update_block) / blocks_between updates) % 2 == 0`, nodes with even
+indices submitBalances right away but only submitPrices after waiting X minutes. On an odd update,
+nodes with odd indices submitBalances right away but only submitPrices after waiting 10 minutes.
+Nodes should do the work immediately and hold off on voting, to avoid unnecessary delay when
+submitting after their 10-minute wait. Note: consensus currently requires more than half of votes,
+so this approach will often mean we need to hit the 10-minute wait before consensus is possible.
+
+### Subcommittee for reward snapshot duty
+This change can be implemented now; however, trees currently take a long time and the subcommittee
+approach works using generous buffers. As a result, it's recommended that this change be implemented
+after trees are reliably fast enough in the watchtower such that this approach doesn't degrade NO
+experience significantly.
+
+Each oDAO node figures out its own index (similar code already exists). On an even rewards period,
+nodes with even indices submitBalances right away but only submitPrices after waiting 30 minutes.
+On an odd update, nodes with odd indices submitBalances right away but only submitPrices after
+waiting X minutes. Nodes should do the work immediately and hold off on voting, to avoid unnecessary
+delay when submitting after their 30-minute wait. Note: consensus currently requires more than half
+of votes, so this approach will often mean we need to hit the 10-minute wait before consensus is
+possible.
+
+It should be explicitly noted that isolated failures are not cause for alarm, and there are simply
+not many data points. The oDAO may wish to conduct an internal test with a member that didn't submit
+when in the active subcommittee.
+
+### Voting past consensus
+This change requires a smart contract release.
+
+It's near-certain that there is a more gas-optimal intervention here; in particular, some conditions
+are nearly-identically checked twice when hitting the execution threshold.
+
+#### submitPrices()
+- Change > to >= in `require(_block > getPricesBlock(), "Network prices for an equal or higher block are set");`
+  - Update message to say "for a higher block"
+- Change `updatePrices(_block, _rplPrice);` to `if (_block != getPricesBlock()) {updatePrices(_block, _rplPrice);}`
```

Comment:
Instead of doing the check for `_block != getPricesBlock()` where `updatePrices` is, It would be more efficient to exit early after the event is emitted with:

```
        if (_block == getPricesBlock()) {
            return;
        }
```